sample/UrlGet/UrlGet.xcodeproj/XCHammerAssets/bazel_build_settings.py

```diff
@@ -0,0 +1,289 @@
+
+    self.debug = debug
+    self.release = release
+
+  def flags(self, is_debug):
+    """Returns the proper flags (either debug or release)."""
+    return self.debug if is_debug else self.release
+
+
+class BazelBuildSettings(object):
+  """Represents a Tulsi project's Bazel settings."""
+
+  def __init__(self, bazel, bazelExecRoot,
+               defaultPlatformConfigId, platformConfigFlags,
+               swiftTargets,
+               cacheAffecting, cacheSafe,
+               swiftOnly, nonSwiftOnly,
+               swiftFeatures, nonSwiftFeatures,
+               projDefault, projTargetMap):
+    self.bazel = bazel
+    self.bazelExecRoot = bazelExecRoot
+    self.defaultPlatformConfigId = defaultPlatformConfigId
+    self.platformConfigFlags = platformConfigFlags
+    self.swiftTargets = swiftTargets
+    self.cacheAffecting = cacheAffecting
+    self.cacheSafe = cacheSafe
+    self.swiftOnly = swiftOnly
+    self.nonSwiftOnly = nonSwiftOnly
+    self.swiftFeatures = swiftFeatures
+    self.nonSwiftFeatures = nonSwiftFeatures
+    self.projDefault = projDefault
+    self.projTargetMap = projTargetMap
+
+  def features_for_target(self, target, is_swift_override=None):
+    """Returns an array of enabled features for the given target."""
+
+    target = _StandardizeTargetLabel(target)
+    is_swift = target in self.swiftTargets
+    if is_swift_override is not None:
+      is_swift = is_swift_override
+
+    return self.swiftFeatures if is_swift else self.nonSwiftFeatures
+
+  def flags_for_target(self, target, is_debug,
+                       config, is_swift_override=None):
+    """Returns (bazel, startup flags, build flags) for the given target."""
+
+    target = _StandardizeTargetLabel(target)
+    target_flag_set = self.projTargetMap.get(target)
+    if not target_flag_set:
+      target_flag_set = self.projDefault
+
+    is_swift = target in self.swiftTargets
+    if is_swift_override is not None:
+      is_swift = is_swift_override
+    lang = self.swiftOnly if is_swift else self.nonSwiftOnly
+
+    config_flags = self.platformConfigFlags[config]
+    cache_affecting = self.cacheAffecting.flags(is_debug)
+    cache_safe = self.cacheSafe.flags(is_debug)
+    target = target_flag_set.flags(is_debug)
+    lang = lang.flags(is_debug)
+
+    startupFlags = []
+    startupFlags.extend(target.startup)
+    startupFlags.extend(cache_safe.startup)
+    startupFlags.extend(cache_affecting.startup)
+    startupFlags.extend(lang.startup)
+
+    buildFlags = []
+    buildFlags.extend(target.build)
+    buildFlags.extend(config_flags)
+    buildFlags.extend(cache_safe.build)
+    buildFlags.extend(cache_affecting.build)
+    buildFlags.extend(lang.build)
+
+    return (self.bazel, startupFlags, buildFlags)
+
+# Default value in case the template does not behave as expected.
+BUILD_SETTINGS = None
+
+BUILD_SETTINGS = BazelBuildSettings(
+    '/Users/jerry/Projects/xchammer-github/sample/UrlGet/tools/bazelwrapper',
```

Comment:
Absolute paths?

Reply:
Yep - this is a code generated asset under the Xcode proj. If we want to do remote caching of `xcodeproj`, this can't be part of that cache, or the `execRoot` will have to be determined at runtime.